app/javascript/components/project-submissions/components/__tests__/create-form.test.jsx

```diff
@@ -50,6 +50,40 @@ describe('CreateForm', () => {
 
       expect(onClose).toHaveBeenCalledTimes(1);
     });
+
+  });
+
+  describe('When submissions have whitespace.', () => {
+    beforeEach(async() => {
+      render(
+        <ProjectSubmissionContext.Provider value={{ lesson: { has_live_preview: true } }}>
+          <CreateForm
+            onSubmit={onSubmit}
+            userId={userId}
+            onClose={onClose}
+          />
+          ,
+        </ProjectSubmissionContext.Provider>,
+      );
+        
+      await act(async () => {
+        const repoUrlInput = screen.getByTestId('repo-url-field');
+        const livePreviewUrlInput = screen.getByTestId('live-preview-url-field');
+        const isPublicToggle = screen.getByTestId('is-public-toggle-slider');
+
+        fireEvent.change(repoUrlInput, { target: { value: '    https://github.com/repo-url      ' } });
+        fireEvent.change(livePreviewUrlInput, { target: { value: '     https://live-preview.com     ' } });
+        fireEvent.change(isPublicToggle, { target: { checked: false } });
+        fireEvent.click(screen.getByTestId('submit-btn'));
+      });
+    });
+    test('calls onSubmit', () => {
```

Comment:
I think it would read better if we combined the two tests into one. We could then describe what we expect to happen to the white space:

Maybe something like this:
```javascript
test('removes the whitespace and successfully submits', () => {
  expect(onSubmit).toHaveBeenCalledTimes(1);
  expect(screen.getByTestId('success-message').textContent).toEqual('Thanks for Submitting Your Solution!');
});
  ```